lib/queuePopulator/IngestionPopulator.js

```diff
@@ -256,20 +258,22 @@ class IngestionPopulator {
                 this.log.info('no log source provisioned, idling',
                               { zkEndpoint });
             }
-            const newRaftReaders = items.map(token =>
-                    new IngestionReader({
-                        zkClient: this.zkClient,
-                        kafkaConfig: this.kafkaConfig,
-                        bucketdConfig: bucketd,
-                        raftId: token,
-                        logger: this.log,
-                        extensions: [this._extension],
-                        metricsProducer: this._mProducer,
-                        ingestionPath,
-                        qpConfig: this.qpConfig,
-                        s3Config: this.s3Config,
-                    })
-                );
+            const newRaftReaders = items.map(token => {
+                const ingestionReader = new IngestionReader({
+                    zkClient: this.zkClient,
+                    kafkaConfig: this.kafkaConfig,
+                    bucketdConfig: bucketd,
+                    raftId: token,
+                    logger: this.log,
+                    extensions: [this._extension],
+                    metricsProducer: this._mProducer,
+                    ingestionPath,
+                    qpConfig: this.qpConfig,
+                    s3Config: this.s3Config,
+                    bucket: this.bucket,
```

Comment:
`bucket` should be provided in `bucketd`. If relying on the unique zenko bucket name, should be saved as `bucketd.name`

we won't be able to keep `this.bucket` on `IngestionPopulator` because we will need to add/remove IngestionReader's based on different sources

Reply:
Aren't we making a new `IngestionPopulator` workflow for each separate bucket?